### ISSUE TITLE
OptionalUnit triggers on a default method in an interface

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1176]
The following code should not be flagged:

```kotlin
interface Foo {
  fun onMapClicked(point: Point?) = Unit
}
```

Using RC9.2
